sdk/resourcemanager/Azure.ResourceManager.Core/tests/Scenario/ResourceGroupBuilderTests.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Core.Tests
+{
+    public class ResourceGroupBuilderTests : ResourceManagerTestBase
+    {
+        public ResourceGroupBuilderTests(bool isAsync)
+            : base(isAsync)//, RecordedTestMode.Record)
+        {
+        }
+
+        [TestCase(null)]
+        [TestCase("  ")]
+        [RecordedTest]
+        public void CreateOrUpdate(string value)
+        {
+            Assert.ThrowsAsync<ArgumentException>(async () => _ = await Client.DefaultSubscription.GetResourceGroups().Construct(LocationData.WestUS2).CreateOrUpdateAsync(value));
```

Comment:
Can we add cases that call .Build after construct and assert the properties passed into the construct method make it to the model?

Reply:
Sure, will update